src/test/harness.rs

```diff
@@ -164,15 +158,29 @@ pub fn expect_messages(results: LsResultList, expected: &[&ExpectedMessage]) {
 }
 
 // Initialise the environment for a test.
-pub fn init_env(project_dir: &str) -> (Cache, TestCleanup) {
+pub fn init_env(project_dir: &str) -> (Config, Cache, TestCleanup) {
+    use std::sync::atomic::{AtomicUsize, Ordering};
+
+    lazy_static! {
+        static ref COUNTER: AtomicUsize = AtomicUsize::new(0);
+    }
```

Comment:
I think it would be better to use the test name rather than a number to make a unique target directory - it is less complexity and makes debugging easier

Reply:
This how I initially did this but then realized that each test that uses shared input files needs to explicitly declare target dir and must ensure that this dir is unique which could become cumbersome to track and guarantee with more tests added (unless there's a way to get the test method name from the runtime). For better debugging maybe we could add some logging?